integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/nextmove/NextMoveServiceBus.java

```diff
@@ -103,52 +110,68 @@ public void putMessage(ConversationResource resource) throws NextMoveException {
-            service.sendMessage(queue, msg);
-        } catch (ServiceException | MessageException | JAXBException | IOException e) {
+            serviceBusClient.sendMessage(os.toByteArray(), queue);
+
+        } catch (JAXBException | IOException | MessageException e) {
             log.error("Could not send conversation resource", e);
-            throw new NextMoveException(e);
+            throw new NextMoveException("Caught exception during message sending:", e);
         }
-
     }
 
-    public void getAllMessages() throws NextMoveException {
-
-        ReceiveMessageOptions opts = ReceiveMessageOptions.DEFAULT;
-        opts.setReceiveMode(ReceiveMode.PEEK_LOCK);
-
-        ServiceBusContract service = createContract();
+    public void getAllMessagesRest() {
         boolean messagesInQueue = true;
         while (messagesInQueue) {
-            ArrayList<BrokeredMessage> messages = Lists.newArrayList();
+            ArrayList<ServiceBusMessage> messages = Lists.newArrayList();
             for (int i=0; i<props.getNextbest().getServiceBus().getReadMaxMessages(); i++) {
-                try {
-                    BrokeredMessage msg = service.receiveQueueMessage(queuePath, opts).getValue();
-                    if (msg == null || isNullOrEmpty(msg.getMessageId())) {
-                        messagesInQueue = false;
-                        break;
-                    }
-                    log.debug(format("Received message on queue=%s with id=%s", queuePath, msg.getMessageId()));
-                    messages.add(msg);
-
-                } catch (ServiceException e) {
-                    log.error("Failed to fetch new message", e);
+                Optional<ServiceBusMessage> msg = serviceBusClient.receiveMessage();
+                if (!msg.isPresent()) {
+                    messagesInQueue = false;
+                    break;
                 }
+                messages.add(msg.get());
             }
 
-            for (BrokeredMessage msg : messages) {
+            for (ServiceBusMessage msg : messages) {
                 try {
-                    Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
-                    EduDocument eduDocument = ((JAXBElement<EduDocument>) unmarshaller.unmarshal(msg.getBody())).getValue();
-                    Optional<ConversationResource> cr = nextMoveQueue.enqueueEduDocument(eduDocument);
+                    Optional<ConversationResource> cr = nextMoveQueue.enqueueEduDocument(msg.getBody());
                     cr.ifPresent(this::sendReceipt);
-                    service.deleteMessage(msg);
-                } catch (JAXBException | ServiceException | IOException e) {
+                    serviceBusClient.deleteMessage(msg);
+                } catch (IOException e) {
                     log.error("Failed to put message on local queue", e);
                 }
             }
         }
     }
 
+    public void getAllMessages() throws ServiceBusException, InterruptedException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: com.microsoft.azure.servicebus.primitives.ServiceBusException, java.lang.InterruptedException [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1160)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.InterruptedException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)